base/src/sgpp/base/operation/hash/OperationMultipleEvalPartialDerivativeModBsplineNaive.cpp

```diff
@@ -0,0 +1,99 @@
+ * use, please see the copyright notice provided with SG++ or at
+ * sgpp.sparsegrids.org
+ *
+ * OperationMultipleEvalPartialDerivativeModBsplineNaive.cpp
+ *
+ * Author: Paul Sarbu
+ */
+
+#include <sgpp/base/operation/hash/OperationMultipleEvalPartialDerivativeModBsplineNaive.hpp>
+
+#include <sgpp/globaldef.hpp>
+
+namespace sgpp {
+namespace base {
+
+void OperationMultipleEvalPartialDerivativeModBsplineNaive::mult(DataVector& alpha,
+                                                                 DataVector& result) {
+  const size_t n = storage.getSize();
+  const size_t d = storage.getDimension();
+  const size_t m = dataset.getNrows();
+
+  result.setAll(0.0);
+
+  pointsInUnitCube = dataset;
+  storage.getBoundingBox()->transformPointsToUnitCube(pointsInUnitCube);
+
+  const double innerDerivative = 1.0 / storage.getBoundingBox()->getIntervalWidth(derivDim);
+
+  for (size_t j = 0; j < m; ++j) {
+    for (size_t i = 0; i < n; ++i) {
+      const GridPoint& gp = storage[i];
+      double curPartialDerivative = 1.0;
+
+      const double dx1d =
+          base.evalDx(gp.getLevel(derivDim), gp.getIndex(derivDim), pointsInUnitCube(j, derivDim)) *
+          innerDerivative;
+
+      for (size_t t = 0; t < d; ++t) {
+        const double val1d = base.eval(gp.getLevel(t), gp.getIndex(t), pointsInUnitCube(j, t));
+
+        if ((val1d == 0.0) && (t != derivDim)) {
+          curPartialDerivative = 0.0;
+          break;
+        }
+
+        curPartialDerivative *= ((t == derivDim) ? dx1d : val1d);
+      }
+
+      result[j] += alpha[i] * curPartialDerivative;
+    }
+  }
+}
+
+void OperationMultipleEvalPartialDerivativeModBsplineNaive::multTranspose(DataVector& alpha,
+                                                                          DataVector& result) {
+  const size_t n = storage.getSize();
+  const size_t d = storage.getDimension();
+  const size_t m = dataset.getNrows();
+
+  result.setAll(0.0);
+
+  pointsInUnitCube = dataset;
+  storage.getBoundingBox()->transformPointsToUnitCube(pointsInUnitCube);
+
+  const double innerDerivative = 1.0 / storage.getBoundingBox()->getIntervalWidth(derivDim);
+
+  for (size_t i = 0; i < n; ++i) {
+    const GridPoint& gp = storage[i];
+
+    for (size_t j = 0; j < m; ++j) {
+      double curPartialDerivative = 1.0;
+
+      const double dx1d =
+          base.evalDx(gp.getLevel(derivDim), gp.getIndex(derivDim), pointsInUnitCube(j, derivDim)) *
+          innerDerivative;
+
+      for (size_t t = 0; t < d; ++t) {
+        const double val1d = base.eval(gp.getLevel(t), gp.getIndex(t), pointsInUnitCube(j, t));
+
+        if ((val1d == 0.0) && (t != derivDim)) {
+          curPartialDerivative = 0.0;
+          break;
+        }
+
+        curPartialDerivative *= ((t == derivDim) ? dx1d : val1d);
+      }
+
+      result[i] += alpha[j] * curPartialDerivative;
+    }
+  }
+}
+
+double OperationMultipleEvalPartialDerivativeModBsplineNaive::getDuration() { return 0.0; }
```

Comment:
What's the purpose of this function and why return zero? Do other operations handle it like this as well?

Reply:
All of the existing MultipleEval classes have that. I haven't quite looked into the purpose of it, I just wanted to have the same unified handling for this operation as well. Maybe the avx/ocl etc. use an actual value there. Not sure of the original purpose, so we might open a separate issue on that.